### PR TITLE
[#8599] [Platform][API] Tablet Server Controller Error handling

### PR DESCRIPTION
**Pull request by mahendranbhat**
Description:

Use new error handling in Tablet Server controller. 

Test Plan:

Its mostly refactor. Unit tests are updated.
compile and runPlatform should work. 

sbt test:

```[error] 	com.yugabyte.yw.controllers.SwaggerGenTest
[error] 	com.yugabyte.yw.commissioner.tasks.RemoveNodeFromUniverseTest
[error] (Test / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 2518 s (41:58), completed 8 Jun, 2021 10:45:32 PM
mahebhat@USBGOMAHEBHAT1 managed % sbt "testOnly com.yugabyte.yw.commissioner.tasks.RemoveNodeFromUniverseTest"
[info] welcome to sbt 1.3.13 (AdoptOpenJDK Java 1.8.0_292)
[info] loading global plugins from /Users/mahebhat/.sbt/1.0/plugins
[info] loading settings for project managed-build from plugins.sbt ...
[info] loading project definition from /Users/mahebhat/Projects/yugabyte/yugabyte-db/managed/project
[info] loading settings for project root from ui-build.sbt,build.sbt ...
[info] set current project to yugaware (in build file:/Users/mahebhat/Projects/yugabyte/yugabyte-db/managed/)
[Yugabyte sbt log] [Resolver] Maven cache server (such as Nexus or Artifactory), specified by YB_MVN_CACHE_URL: List()
[Yugabyte sbt log] [Resolver] Local resolver (enabled by USE_MAVEN_LOCAL, path can be customized with YB_MVN_LOCAL_REPO): List(cache:Maven2 Local: /Users/mahebhat/.m2/repository)
[Yugabyte sbt log] [Resolver] Default resolver: Vector(FileRepository(local, Patterns(ivyPatterns=Vector(${ivy.home}/local/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)([branch]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), artifactPatterns=Vector(${ivy.home}/local/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)([branch]/)[revision]/[type]s/[artifact](-[classifier]).[ext]), isMavenCompatible=false, descriptorOptional=false, skipConsistencyCheck=false), FileConfiguration(true, None)), public: https://repo1.maven.org/maven2/)
[Yugabyte sbt log] [Resolver] Snapshot resolver for yb-client jar (used when USE_MAVEN_LOCAL is not set, mostly during local development, configured with YB_MVN_SNAPSHOT_URL): List()
[info] Instrumenting 743 classes to /Users/mahebhat/Projects/yugabyte/yugabyte-db/managed/target/scala-2.12/jacoco/instrumented-classes
[info] Test run started
[info] Test com.yugabyte.yw.commissioner.tasks.RemoveNodeFromUniverseTest.testRemoveNodeSuccess started
[info] Test com.yugabyte.yw.commissioner.tasks.RemoveNodeFromUniverseTest.testRemoveNodeWithMaster started
[info] Test com.yugabyte.yw.commissioner.tasks.RemoveNodeFromUniverseTest.testRemoveNonExistentNode started
[info] Test com.yugabyte.yw.commissioner.tasks.RemoveNodeFromUniverseTest.testRemoveNodeRF5 started
[info] Test com.yugabyte.yw.commissioner.tasks.RemoveNodeFromUniverseTest.testRemoveNodeWithNoDataMoveRF5 started
[info] Test com.yugabyte.yw.commissioner.tasks.RemoveNodeFromUniverseTest.testRemoveNodeWithNoDataMove started
[info] Test com.yugabyte.yw.commissioner.tasks.RemoveNodeFromUniverseTest.testRemoveUnknownNode started
[info] Test run finished: 0 failed, 0 ignored, 7 total, 26.564s
[info] Passed: Total 7, Failed 0, Errors 0, Passed 7
[success] Total time: 30 s, completed 8 Jun, 2021 11:01:52 PM
```

### CODE REVIEW

File: managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java [hunk 1]
```diff
@@ -52,7 +53,7 @@ public Result list() {
                 tabletServers.add(tabletServer.getHost());
               });
     } catch (Exception e) {
```
Comment (mahendranbhat) [inline managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java:55]: We can not remove this try catch , right?

File: managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java [hunk 2]
```diff
@@ -86,11 +87,10 @@ public Result listTabletServers(UUID customerUUID, UUID universeUUID) {
       final int masterHttpPort = universe.getUniverseDetails().communicationPorts.masterHttpPort;
       final String masterLeaderUrl =
           String.format("http://%s:%s/api/v1/tablet-servers", masterLeaderIPAddr, masterHttpPort);
-
       response = apiHelper.getRequest(masterLeaderUrl);
     } catch (Exception e) {
```
Comment (mahendranbhat) [inline managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java:91]: Can we remove this?

File: managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java (hunk 1 above)
Comment (sb-yb) [reply managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java:55]: You can have try with only finally. 
So remove this.

File: managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java (hunk 2 above)
Comment (sb-yb) [reply managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java:91]: Maybe. But not sure if the error will be clear enough in that case. Lets keep it.

**Review by sb-yb — COMMENTED** [review]
Responded to questions

File: managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java (hunk 2 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java:91]: There is a log too and I think we need the log so keeping it.

File: managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java (hunk 1 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java:55]: we need to catch this as `client.listTabletServers();` throwing an exception .

Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/controllers/TabletServerController.java:55]: or I can change the method signature

**Review by sb-yb — APPROVED** [review]

**Pull request merged**